pkg/nuclide/flow/lib/FlowLinter.js

```diff
@@ -30,39 +30,42 @@ var {JS_GRAMMARS} = require('./constants.js');
  * with which the usage disagrees. Note that these could occur in different
  * files.
  */
-function flowMessageToLinterMessage(message) {
+function extractRange(message){
   // It's unclear why the 1-based to 0-based indexing works the way that it
   // does, but this has the desired effect in the UI, in practice.
-  var range = new Range(
+  return new Range(
     [message['line'] - 1, message['start'] - 1],
     [message['endline'] - 1, message['end']]
   );
+}
 
-  return {
-    type: 'Error',
-    text: message['descr'],
-    filePath: message['path'],
-    range: range,
-  };
+// A trace object is very similar to an error object.
+function flowMessageToTrace(message){
+  return { 
+    type: 'Trace',
+    text: message.descr,
+    filePath: message.path,
+    range: extractRange(message)
+  }
 }
 
-/**
- * In some cases, flow diagnostics will span multiple files. It helps in the case
- * that there's a problem with, say, the the way a function defined in another
- * file is typed conflicts with how you're calling it.
- *
- * You get diagnostics like:
- * File A: <Type string is incompatable>
- * File B: <with type number>
- *
- * We don't have any way to deal with this, so merge the descriptions, so that
- * information doesn't get cut off.
- *
- */
-function mergeFlowMessages(messages: Array) {
-  var message = messages[0];
-  message['descr'] = messages.map((msg)=>msg['descr']).join(' ');
-  return message;
+function flowMessageToLinterMessage(arr) {
+  var message = arr[0]
+
+  var obj = { 
+    type: message.level,
+    text: arr.map(obj => obj.descr).join(' '),
```

Comment:
Since `obj` is already defined in this scope, please use a different variable name here to avoid shadowing.